src/fe-common/core/fe-channels.c

```diff
@@ -625,6 +630,7 @@ void fe_channels_init(void)
 {
 	settings_add_bool("lookandfeel", "autoclose_windows", TRUE);
 	settings_add_bool("lookandfeel", "show_names_on_join", TRUE);
+	settings_add_int("lookandfeel", "show_names_on_join_limit", 18);
```

Comment:
Yeah, nah, this won't do, I'm going to bikeshed the fuck out of this constant.

Reply:
which constant would you prefer?

Reply:
n.b this is 3 lines (3 × 6)

Reply:
I guess I just disagree with the whole thing of "having too many nicks". If a channel is huge, the flood of names on join is useful information for me.

Default disabled?

Reply:
I just think we should enable this. you still get the names count if the limit is exceeded and can easily type /n if you really need to see the list

but most of the time I'd argue the full list is useless on large channels

Reply:
the full names list regularly hides the topic message and other more useful information

Reply:
it's also a compromise to turning off show_names_on_join by default

Reply:
I'm ok with the feature existing, but if people have a preference for a maximum number of nicks that are worth showing, they can set it in the setting. I don't like sneaky changes of default behavior, and I wouldn't want this changed in my own config.

Reply:
I understand your point, but I think we should change the default settings somehow. The defaults represent what the user sees by default and I think we should make that better somehow.

I've opened #1276 to discuss this